packages/dev/core/src/Meshes/abstractMesh.hotSpot.ts

```diff
@@ -140,30 +110,27 @@ export function GetHotSpotToRef(mesh: AbstractMesh, hotSpotQuery: HotSpotQuery,
 
     // compute normal in world space
     if (resNormal) {
-        resNormal.set(0, 0, 0);
+        const pointA = TmpVectors.Vector3[0];
+        const pointB = TmpVectors.Vector3[1];
+        const pointC = TmpVectors.Vector3[2];
+        const segmentA = TmpVectors.Vector3[3];
+        const segmentB = TmpVectors.Vector3[4];
+        segmentA.copyFrom(pointB);
+        segmentA.subtractInPlace(pointA);
+        segmentB.copyFrom(pointC);
+        segmentB.subtractInPlace(pointA);
+        segmentA.normalize();
+        segmentB.normalize();
+        Vector3.CrossToRef(segmentB, segmentA, resNormal);
 
-        if (mesh.getVerticesData(VertexBuffer.NormalKind)) {
-            // use normal buffer
-            for (let i = 0; i < 3; i++) {
-                const index = hotSpotQuery.pointIndex[i];
-                GetTransformedNormal(mesh, index, TmpVectors.Vector3[0]);
-                TmpVectors.Vector3[0].scaleAndAddToRef(hotSpotQuery.barycentric[i], resNormal);
-            }
-        } else {
-            // estimate normal
-            const pointA = TmpVectors.Vector3[0];
-            const pointB = TmpVectors.Vector3[1];
-            const pointC = TmpVectors.Vector3[2];
-            const segmentA = TmpVectors.Vector3[3];
-            const segmentB = TmpVectors.Vector3[4];
-            segmentA.copyFrom(pointB);
-            segmentA.subtractInPlace(pointA);
-            segmentB.copyFrom(pointC);
-            segmentB.subtractInPlace(pointA);
-            segmentA.normalize();
-            segmentB.normalize();
-            Vector3.CrossToRef(segmentA, segmentB, resNormal);
+        // flip normal
+        let invertWinding = mesh.material && mesh.material.sideOrientation === Constants.MATERIAL_CounterClockWiseSideOrientation;
+        invertWinding ||= mesh.getWorldMatrix().determinant() < 0;
+        invertWinding ||= mesh.getScene().useRightHandedSystem;

```

Comment:
What if two of the 3 conditions (`sideOrientation==CCW`, `determinant < 0` and `RHS`) are true at the same time? Do they cancel each other and the winding should not be inverted?

Reply:
Merging for now, I'll continue digging this and will open a new PR if needed.